File: little-bee/controller/router/signal_unification.go
package router

import (
	"controller/config"

	"github.com/gin-gonic/gin"
)

const URL_ROUTER_SIGNAL_UNIFICATION = "/config/signal-unification"
const URL_ROUTER_SIGNAL_UNIFICATION_BY_ID = "/config/signal-unification/:signal-unification-id"
const URL_ROUTER_SIGNAL_UNIFICATION_QUERY = "/config/signal-unification/query"
const URL_ROUTER_SIGNAL_UNIFICATION_ALL = "/config/signal-unification/all"

func SignalUnificationRouter(r *gin.RouterGroup) *gin.RouterGroup {
	r.POST(URL_ROUTER_SIGNAL_UNIFICATION, config.CreateSignalUnification)
	r.GET(URL_ROUTER_SIGNAL_UNIFICATION_BY_ID, config.RetrieveSignalUnification)
	r.PUT(URL_ROUTER_SIGNAL_UNIFICATION, config.UpdateSignalUnification)
	r.DELETE(URL_ROUTER_SIGNAL_UNIFICATION_BY_ID, config.DeleteSignalUnification)
	r.PUT(URL_ROUTER_SIGNAL_UNIFICATION_BY_ID, config.DeleteSignalUnificationWithFlag)
	r.POST(URL_ROUTER_SIGNAL_UNIFICATION_QUERY, config.QuerySignalUnification)
	r.GET(URL_ROUTER_SIGNAL_UNIFICATION_ALL, config.GetAllSignalUnifications)
	return r
}

File: frontend/console/src/api/signal-unification.js
import Axios from "@/http";

export function createSignalUnification(params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/signal-unification', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function retrieveSignalUnification(signalUnificationId) {
    return new Promise((resolve, reject) => {
        Axios.get(`/config/signal-unification/${ signalUnificationId }`).then(res => {
            resolve(res)
        })
    })
}

export function updateSignalUnification(params) {
    return new Promise((resolve, reject) => {
        Axios.put('/config/signal-unification', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function deleteSignalUnification(signalUnificationId) {
    return new Promise((resolve, reject) => {
        Axios.delete(`/config/signal-unification/${ signalUnificationId }`).then(res => {
            resolve(res)
        })
    })
}

export function deleteSignalUnificationWithFlag(signalUnificationId) {
    return new Promise((resolve, reject) => {
        Axios.put(`/config/signal-unification/${ signalUnificationId }`).then(res => {
            resolve(res)
        })
    })
}

export function getAllSignalUnifications() {
    return new Promise((resolve, reject) => {
        Axios.get('/config/signal-unification/all').then(res => {
            resolve(res)
        })
    })
}

export function querySignalUnification (params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/signal-unification/query', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}